pages/index.js

```diff
@@ -1,42 +1,40 @@
 import Head from "next/head";
-import React, { useRef, useEffect, useState } from "react";
-import "mapbox-gl/dist/mapbox-gl.css";
-import mapboxgl from "!mapbox-gl"; // eslint-disable-line import/no-webpack-loader-syntax
-import styled from "styled-components";
-
-export function getServerSideProps() {
-  return {
-    props: {
-      accessToken: process.env.MAPBOX_TOKEN,
-    },
-  };
-}
-
-export default function Home({ accessToken }) {
-  const mapContainer = useRef(null);
-  const map = useRef(null);
-  const [lng, setLng] = useState(-70.9);
-  const [lat, setLat] = useState(42.35);
+import { useState } from "react";
+import Map from "@/components/Map/Map";
+import MapWrapper from "@/components/MapWrapper/StyledMapWrapper";
+import SearchPlaces from "@/components/SearchPlaces/SearchPlaces";
+
+// Change Token before git push!!!
+const accessToken =
+  "pk.eyJ1Ijoia2V0bmVrIiwiYSI6ImNsaDVybDl5azA0ZzYzZ21wcmJvZnlxeGYifQ.7xOZhYeVEcLNK833qEUjUw";
+
+export default function Home() {
+  const [lng, setLng] = useState(10.000654);
+  const [lat, setLat] = useState(53.550341);
   const [zoom, setZoom] = useState(9);
+  const [center, setCenter] = useState(null);
+
+  const getNewDestination = async (url) => {
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        console.error("Loading new destination went wrong!");
+      }
+      const data = await response.json();
+      setCenter(data?.features[0]?.center);
+    } catch (error) {
+      console.error(error);
+    }
+  };
 
-  mapboxgl.accessToken = accessToken;
-
-  useEffect(() => {
-    map.current = new mapboxgl.Map({
-      container: mapContainer.current,
-      style: "mapbox://styles/mapbox/streets-v12",
-      center: [lng, lat],
-      zoom: zoom,
-    });
-  }, []);
-
-  useEffect(() => {
-    map.current.on("move", () => {
-      setLng(map.current.getCenter().lng.toFixed(4));
-      setLat(map.current.getCenter().lat.toFixed(4));
-      setZoom(map.current.getZoom().toFixed(2));
-    });
-  }, []);
+  const handleSearchSubmit = (event) => {
+    event.preventDefault();
+    const form = event.target;
+    const inputValue = form.elements[1].value;
+    getNewDestination(
+      `https://api.mapbox.com/geocoding/v5/mapbox.places/${inputValue}.json?proximity=ip&access_token=${accessToken}`
```

Comment:
We checked a few searches: Münster, Kamen, Hamburg …
These led to unexpected locations, might be due to the "proximity=ip" parameter.
You might want to check the results without that parameter.

Reply:
I removed "proximity=ip". Now it works better but when I want to find Kamen I have to chain Germany behind, because I always use the first result from the array that I get from the api and Kamen comes on the fours place I guess. But with my next story I will implement a result window where the user can choose from given suggestions and than he can click on Kamen Germany😊